File: Functions/financial_calc_update.py
# Tatiana Susov, Financial Calculator Python

def info(income, amount, type):
    pertype = amount/income*100
    print(f"You spend ${amount:.2f} on {type} and that is {pertype:.0f}% of your income.")
   

print("Welcome to your personal financial/budget calculator, lets begin!")

def price(price):
    return float(input(f"what is your monthly spending on {price}? "))

income = price("income")
rent = price("rent")
utilities = price("utilities")
groceries = price("groceries")
transportation = price("transportation")
savings = income/10
total_spending = income-(rent+utilities+groceries+transportation+savings)
info(income, rent, "rent")
info(income, utilities, "utilities")
info(income, groceries, "groceries")
info(income, transportation, "transportation")
info(income, savings, "savings")
info(income, total_spending, "everything in total")

File: Functions/financial_calc_update.c
//Tatiana Susov, Update Financial Calculator C
#include <stdio.h>
#include <math.h>

int info(float price, float income, char type[50]){
   float percent = price/income*100;
   printf("The amount of money that goes to %s is $%.2f, which is %.1f%% of your income.\n", type, price, percent);
}
float userinformation(char charge[20]){
   float answer;
   printf("How much money do you spend on %s?\n", charge);
   scanf("%f", &answer);
   return answer;
}
float income;
int main(void){
   printf("%s\n", "Welcome to your personal financial/budget calculator, lets begin!");
   printf("What is your monthly income?:");
   scanf("%f", &income);
   float savings = income/10;
   float rent = userinformation("rent");
   float utilities = userinformation("utilities");
   float groceries = userinformation("groceries");
   float transportation = userinformation("transportation");
   float spending = (income-rent-utilities-groceries-transportation-savings);
   info(rent, income, "rent");
   info(utilities, income, "utilities");
   info(groceries, income, "groceries");
   info(transportation, income, "transportation");
   info(savings, income, "savings");
   info(spending, income, "spending");
    return 0;
}